Tasks/AzureRmWebAppDeployment/azurermutil.ts

```diff
@@ -129,25 +131,30 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
     return deferred.promise;
 }
 
-async function getAzureRMWebAppID(SPN, webAppName: string, resourceType: string) {
+async function getAzureRMWebAppID(SPN, webAppName: string,url: string) {
 
     var deferred = Q.defer<any>();
     var accessToken = await getAuthorizationToken(SPN);
```

Comment:
Access token shouldn't be fetched for each call, do this outside. Verify access token expiration time as well.